pkg/generate/app/cmd/newapp.go

```diff
@@ -710,20 +711,146 @@ func (c *AppConfig) Run() (*AppResult, error) {
 	}, nil
 }
 
+// followRefToDockerImage follows a buildconfig...To/From reference until it terminates in docker image information.
+// This can include dereferencing chains of ImageStreamTag references that already exist or which are being created.
+// ref is the reference to To/From to follow. If ref is an ImageStreamTag that is following another ImageStreamTag, isContext
+// should be set to the parent IS. Finally, objects is the list of objects that new-app is creating to support the buildconfig.
+// It returns a reference to a terminal DockerImage or nil if one could not be determined (a valid, non-error outcome). err
+// is only used to indicate that the follow encountered a severe error (e.g malformed data).
+func (c *AppConfig) followRefToDockerImage(ref *kapi.ObjectReference, isContext *imageapi.ImageStream, objects app.Objects) (*kapi.ObjectReference, error) {
+
+	if ref == nil {
+		return nil, fmt.Errorf("Unable to follow nil")
+	}
+
+	if ref.Kind == "DockerImage" {
+		// Make copy so we don't modify any structures new-app/build created
+		copy := *ref
+		// Namespace should not matter here. The DockerImage URL will include project information if it is relevant.
```

Comment:
I'm a bit freak with tidiness, when I look at this I see 3 blocks of comments in 3 different styles.
The first misses a final stop, while the other two have.
The second is a long line, while the others are apparently reflowed to fit 80 columns.
And the third looks tidy -- full sentences nicely formatted.

Ah, and the commentary above the function would also benefit from reflowing.

Since we're not normally editing this type of thing later on, I think we should try to be at least consistent with new code we add :)
But again, that might be just me.


--------
https://github.com/golang/go/wiki/CodeReviewComments#line-length
https://golang.org/doc/effective_go.html#commentary